### PR TITLE
update react-leaflet to 2.0.0

### PR DESCRIPTION
**Pull request by anajavi**
This PR adds support for react-leaflet@2.0.0.

Also drops support for React < 16.3, the same as react-leaflet@2.

Updates peer dependency leaflet-draw to 1.0.2 .

I'm happy to change this, if there are problems with it.

### CODE REVIEW

**Comment by sanfilippopablo** [comment]
Any update on this?

**Comment by anajavi** [comment]
@sanfilippopablo would you like to test this PR? It works for me, but it would be nice if somebody else tries it out too.

**Comment by anajavi** [comment]
I would also like to drop leaflet-draw < 1.0.0 from peerdependencies. Any objections?

**Comment by sanfilippopablo** [comment]
I tried this. It worked, but only after I copied the EditControl's code to my project and imported it from there. Building an importing the project gave me errors. I'm going to test it more with more time later.

**Comment by FeNoMeNa** [comment]
Any progress?

**Comment by anajavi** [comment]
I consider it ready.

@alex3165 do you think this can be merged?

**Comment by batazor** [comment]
@alex3165 Any progress?

**Comment by rfestag** [comment]
@alex3165 I'm also interested in this. Anything stopping the merge?

File: package.json [hunk 1]
```diff
@@ -55,12 +55,12 @@
     "eslint": "^3.8.1",
     "eslint-plugin-react": "^6.4.1",
     "json-loader": "^0.5.4",
-    "leaflet": "^1.0.3",
-    "leaflet-draw": "^0.4.9",
+    "leaflet": "^1.3.1",
+    "leaflet-draw": "^1.0.2",
```
Comment (aqueiros) [inline package.json]: I believe this version of leaflet-draw is broken:
https://github.com/Leaflet/Leaflet.draw/issues/872

However, a new 1.0.3 tag was released 6 days ago that fixes this. It should probably use that one instead.

Comment (anajavi) [reply package.json]: Updated to leaflet-draw@1.0.3

**Comment by aqueiros** [comment]
Hey! I have been trying to test this PR but I am unable to make it work. I have a working environment with a simple example:

my package.json:
`
    "leaflet": "^1.3.4",
    "leaflet-draw": "^1.0.3",
    "react-leaflet": "^2.0.1",
    "react-leaflet-draw": "file:../react-leaflet-draw",  #--> I cloned this branch to my local
`
screenshot 2018-09-25 11 28 24

```
export default class MapPage extends React.Component {
  state = {
    lat: 51.505,
    lng: -0.09,
    zoom: 13,
  };
  render() {
    const position = [this.state.lat, this.state.lng];
    return (
      <Map id="mymap" center={position} zoom={this.state.zoom}>
        <TileLayer
          attribution="&amp;copy <a href=&quot;http://osm.org/copyright&quot;>OpenStreetMap</a> contributors"
          url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
        />
        <Marker position={position}>
          <Popup>
            A pretty CSS3 popup. <br /> Easily customizable.
          </Popup>
        </Marker>
      </Map>
    );
  }
}
```

My first piece of problem is: as soon as I add `import { EditControl } from 'react-leaflet-draw';`
to this file, I immediately get a problem 

`Uncaught TypeError: Cannot read property 'Marker' of undefined`. But I just ignored this. I removed the Marker code and then it was fine.

But then, when I add the EditControl to the map, following [this example](https://github.com/alex3165/react-leaflet-draw/blob/master/example/edit-control.js)
I get this error:

`react-leaflet-draw.js:308 Uncaught TypeError: _leaflet.Control.Draw is not a constructor
    at createDrawElement (react-leaflet-draw.js:308)
    at EditControl.createLeafletElement (react-leaflet-draw.js:203)
    at EditControl.MapControl (MapControl.js:25)
    at new EditControl (react-leaflet-draw.js:191)
    at constructClassInstance (react-dom.development.js:11447)
    at updateClassComponent (react-dom.development.js:13144)
    at beginWork (react-dom.development.js:13824)
    at performUnitOfWork (react-dom.development.js:15863)
    at workLoop (react-dom.development.js:15902)
    at HTMLUnknownElement.callCallback (react-dom.development.
`

Also, before trying to install this locally, I used this:
"react-leaflet-google": "github:alex3165/react-leaflet-draw#pull/49/head",
which installed the package in my node_modules, but it was unable to create a /dist folder and therefore my project was unable to import it. I used the same method to test [this PR from react-leaflet-google](https://github.com/Charmatzis/react-leaflet-google/pull/29/files) and everything worked out fine.

Am I missing something?  Can I get some instructions on how to do this properly, since there are some people that say they managed to test this locally? 
Thanks!

**Comment by anajavi** [comment]
Putting this into package.json seems to make module loading hairy:
"react-leaflet-draw": "file:../react-leaflet-draw"

The easiest way to test is to do normal npm install react-leaflet-draw, and then replace the node_modules/dist/react-leaflet-draw.js with newly built version. npm will whine about peerDependencies, but those can be ignored.

**Comment by aqueiros** [comment]
Thank you @anajavi , but I still keep having problems.

1. I installed the latest react-leaflet-draw on my test project
2. I ran ```npm run build:dist``` on my local react-leaflet-draw cloned repo (pointing to this branch)
3. I replaced the contents of the generated dist into _my_test_project/node_modules/react-leaflet-draw/dist_ . 
4. Added  ```import { EditControl } from 'react-leaflet-draw'``` to my test component

Now I am at a point where even removing the <Map> component, as soon as I add this line 
``` import { EditControl } from 'react-leaflet-draw'; ``` to my test component, I get this error:

```
react-leaflet-draw.js:103 Uncaught TypeError: Super expression must either be null or a function, not object
    at _inherits (react-leaflet-draw.js:103)
    at eval (react-leaflet-draw.js:122)
    at Object.eval (react-leaflet-draw.js:228)
    at __webpack_require__ (react-leaflet-draw.js:25)
    at Object.defineProperty.value (react-leaflet-draw.js:59)
    at __webpack_require__ (react-leaflet-draw.js:25)
    at Object.defineProperty.value (react-leaflet-draw.js:45)
    at eval (react-leaflet-draw.js:48)
    at webpackUniversalModuleDefinition (react-leaflet-draw.js:3)
    at eval (react-leaflet-draw.js:5)
```

Is this not the correct way to import the EditControl?

**Comment by srghma** [comment]
@anajavi what about merge this and https://github.com/alex3165/react-leaflet-draw/pull/48 PR's?

**Comment by anajavi** [comment]
@srghma you seem to beat me to it:)

@aqueiros you can also just copy/paste the EditControl source code to your project.

**Pull request merged**